app/assets/javascripts/bindingtarifftraderfrontend.js

```diff
@@ -128,7 +130,11 @@ $(document).ready(function () {
         e.preventDefault();
         if (window.history && window.history.back && typeof window.history.back === 'function' &&
             (docReferrer !== "" && docReferrer.indexOf(window.location.host) !== -1)) {
-            window.history.back();
+            const historyStack = JSON.parse(sessionStorage.getItem("historyStack")) || [];
+            historyStack.pop();    //take the url for page just exited off the stack
+            const previousUrl = historyStack[historyStack.length - 1]
+            window.location.href = previousUrl
+            sessionStorage.setItem("historyStack", JSON.stringify(historyStack));
```

Comment:
This looks good but I'd like to see a little more defensive programming here due to issues like this:

<img width="421" alt="Screenshot 2020-10-08 at 16 21 59" src="https://user-images.githubusercontent.com/2992938/95479249-7481e380-0982-11eb-88ce-73104aea379d.png">

Reply:
I realise that we should always get the current page URL on the stack but I just don't trust JS 😆